frame/fast-unstake/src/lib.rs

```diff
@@ -339,38 +379,42 @@ pub mod pallet {
 				return T::DbWeight::get().reads(2)
 			}
 
-			let UnstakeRequest { stash, mut checked, deposit } =
-				match Head::<T>::take().or_else(|| {
-					// NOTE: there is no order guarantees in `Queue`.
-					Queue::<T>::drain()
-						.map(|(stash, deposit)| UnstakeRequest {
-							stash,
-							deposit,
-							checked: Default::default(),
-						})
-						.next()
-				}) {
-					None => {
-						// There's no `Head` and nothing in the `Queue`, nothing to do here.
-						return T::DbWeight::get().reads(4)
-					},
-					Some(head) => head,
-				};
+			let UnstakeRequest { stashes, mut checked } = match Head::<T>::take().or_else(|| {
+				// NOTE: there is no order guarantees in `Queue`.
```

Comment:
Is there a size limit on `Queue`? I see that it is a `CountedStorageMap` but there is no limit?!

Reply:
No, and I don't see a need for it to have any bound either -- we never iterate the whole thing.